sunflower-land/sunflower-land
# [WIP-FEAT] Achievements

# Description

**This feature includes an Achievement system.** 

_You need to complete tasks to earn rewards_

- [ ] Achievement system [70%]
- [X] Harvest Plant Achievement
- [x] Earn Rock/Iron/Gold Achievement
- [ ] Earn wood Achievement
- [ ] Craft Item Achievement
- [ ] Reward system

- [ ] UI (maybe to rework) [80%]
- [ ] Optimization [10%]
- [ ] Achievements redaction [0%]


## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

At the moment, this feature is a work in progress.

Some bugs are highly possible.

To test :
- Go to your farm and check if the achievement menu is placed
- Start to earn sunflowers and check if the progress increases.

If the first load throws errors, that's due to the temporary system for the progress. It's stored in local storage and the initialized might not work the first time.

# Preview

Video : https://teissieryannis.com/assets/img/DEMO-Achievement-V1.mp4

![preview-achievements-gif](https://teissieryannis.com/assets/img/Prew-Achievements.gif)


# Checklist:
